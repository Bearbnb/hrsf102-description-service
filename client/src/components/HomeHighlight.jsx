import React from 'react';
import PropTypes from 'prop-types';

import Thumbs from './Thumbs.jsx';
import Highlight from './Highlight.jsx';

import styles from '../../../styles/HomeHighlight.css';

class HomeHighlight extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showThumbs: true,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({
      showThumbs: false,
    });
  }

  render() {
    const { highlight } = this.props;
    const { showThumbs } = this.state;
    let buttonOrFeedback;

    if (showThumbs) {
      buttonOrFeedback = (
        <Thumbs handleClick={this.handleClick} />
      );
    } else {
      buttonOrFeedback = (
        <div className={styles.helpful}>
          Thanks for your feedback.
        </div>
      );
    }

    return (
      <div className={styles.container}>
        <Highlight highlight={highlight} />
        {buttonOrFeedback}
      </div>
    );
  }
}

HomeHighlight.propTypes = {
  highlight: PropTypes.objectOf(PropTypes.string),
};

HomeHighlight.defaultProps = {
  highlight: {
    highlight: 'Self Check-in',
    details: 'Easily check yourself in with the lockbox.',
  },
};

export default HomeHighlight;
