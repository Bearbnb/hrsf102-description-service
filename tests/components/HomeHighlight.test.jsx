import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import HomeHighlight from '../../client/src/components/HomeHighlight';

describe('<HomeHighlight />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<HomeHighlight />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
