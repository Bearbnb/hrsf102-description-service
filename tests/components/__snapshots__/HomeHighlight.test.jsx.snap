// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeHighlight /> should render properly 1`] = `
<div
  className="container"
>
  <span
    className="highlight"
  >
    Self Check-in
  </span>
   · 
  <span
    className="details"
  >
    Easily check yourself in with the lockbox.
  </span>
  <div
    className="helpful"
  >
    <button
      className="button"
      type="button"
    >
      Helpful
      <i
        className="far fa-thumbs-up"
      />
    </button>
    <span
      className="interpunct"
    >
       · 
    </span>
    <button
      className="button"
      type="button"
    >
      Not Helpful
    </button>
  </div>
</div>
`;
